descriptor.go

```diff
@@ -0,0 +1,30 @@
+package distribution
+
+import "github.com/docker/distribution/digest"
+
+// Descriptor describes targeted content. Used in conjunction with A value can be used to fetch, store
+// and target any kind of blob. The struct also describes the wire protocol
+// format. Fields should only be added but never changed.
+type Descriptor struct {
+	// MediaType describe the type of the content. All text based formats are
+	// encoded as utf-8.
+	MediaType string `json:"mediaType,omitempty"`
+
+	// Size in bytes of content.
+	Size int64 `json:"size,omitempty"`
+
+	// Digest uniquely identifies the content. A byte stream can be verified
+	// against against this digest.
+	Digest digest.Digest `json:"digest,omitempty"`
+
+	// Repository optionally specifies a named repository from which this
+	// content should always be accessed. Discovery should be performed to
+	// resolve the location.
+	Repository string `json:"repository,omitempty"`
```

Comment:
Mmm... I'm trying to understand how this relates to the namespace document that was shared earlier. Isn't the discovery mechanism specific to a "top-level name" (i.e., an image)?